### PR TITLE
feat: IAM auth token generator for RDS

### PR DESCRIPTION
**Pull request by sichanyoo**
## Issue \#

- https://github.com/awslabs/aws-sdk-swift/issues/242
- 2173

## Description of changes

Changes:
- Adds `AWSClientRuntime.AuthTokenGenerator`. It's tagged with `@_spi` to hide it and re-expose through AWSRDS module.
- Adds integration for code-generating`AWSRDS.AuthTokenGenerator`, which is a wrapper class for `AWSClientRutime.AuthTokenGenerator` (this is done to expose token generator via RDS module, as outlined by specifications)

Tested using a dummy AWS RDS MySQL database and confirmed it works.

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/Customizations/RDS/AuthTokenGenerator.swift [hunk 1]
```diff
@@ -0,0 +1,100 @@
+//
```
Comment (sichanyoo) [inline Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/Customizations/RDS/AuthTokenGenerator.swift:1]: This is the handwritten code for token generator, located in AWSClientRuntime

File: Sources/Services/AWSRDS/Sources/AWSRDS/AuthTokenGenerator.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
+//
```
Comment (sichanyoo) [inline Sources/Services/AWSRDS/Sources/AWSRDS/AuthTokenGenerator.swift:1]: This is the generated token generator located in AWSRDS that wraps AWSClientRuntime.AuthTokenGenerator; it's included in the PR as reference for reviewers on what generated output looks like.

File: codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/rds/AuthTokenGeneratorIntegration.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
+/*
```
Comment (sichanyoo) [inline codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/rds/AuthTokenGeneratorIntegration.kt:1]: This is the new codegen integration that writes a new file with wrapper token generator if service is RDS.

File: codegen/smithy-aws-swift-codegen/src/main/resources/META-INF/services/software.amazon.smithy.swift.codegen.integration.SwiftIntegration [hunk 4]
```diff
@@ -26,3 +26,4 @@ software.amazon.smithy.aws.swift.codegen.AWSClientConfigurationIntegration
 software.amazon.smithy.swift.codegen.swiftintegrations.InitialRequestIntegration
 software.amazon.smithy.aws.swift.codegen.swiftintegrations.RegistryConfigIntegration
 software.amazon.smithy.aws.swift.codegen.swiftintegrations.AmzSdkRetryHeadersIntegration
+software.amazon.smithy.aws.swift.codegen.customization.rds.AuthTokenGeneratorIntegration
```
Comment (sichanyoo) [inline codegen/smithy-aws-swift-codegen/src/main/resources/META-INF/services/software.amazon.smithy.swift.codegen.integration.SwiftIntegration:29]: Makes the new integration run during codegen

**Review by sichanyoo — COMMENTED** [review]
Misc. informational comments

**Review by dayaffe — APPROVED** [review]

**Pull request merged**